lib/bullet.js

```diff
@@ -15,47 +15,82 @@ Bullet.prototype.draw = function () {
   this.context.arc(this.x, this.y, this.radius, 2 * Math.PI, false);
   this.context.fillStyle = "#000000";
   this.context.fill();
+
   return this;
 };
 
-Bullet.prototype.move = function (obstacle) {
-  if(this.x > this.context.canvas.width - this.width){
-    this.directionX = -this.directionX;
-    this.bounceCount++;
-  }
-  if(this.x < 0){
-    this.directionX = -this.directionX;
-    this.bounceCount++;
-  }
-  if(obstacle.height > 10 && this.x > obstacle.x - obstacle.width && this.x < obstacle.x + obstacle.width * 2 && this.y >= obstacle.y && this.y < obstacle.y + obstacle.height && obstacle.y > 0){
-    this.directionX = -this.directionX;
-    this.bounceCount++;
-  }
-  if(obstacle.height > 10 && this.x > obstacle.x - obstacle.width && this.x < obstacle.x + obstacle.width * 2 && obstacle.y === 0 && this.y < obstacle.height){
-    this.directionX = -this.directionX;
-    this.bounceCount++;
-  }
+
+Bullet.prototype.move = function (obstacles) {
+  this.collisionCheck(obstacles);
```

Comment:
Pulled out logic for collision detection into it's own function, so the `move` function isn't so "dirty"